### PR TITLE
dynamic set the retry for retryable error on worker processor

### PR DESCRIPTION
**Pull request by wxing1292**

### CODE REVIEW

File: service/worker/processor.go [hunk 1]
```diff
@@ -65,6 +94,12 @@ type (
 		metricsClient    metrics.Client
 		domainReplicator DomainReplicator
 		historyClient    history.Client
+
+		sync.Mutex
+		// worker in retry is used by underlying processor when doing retry on a task
+		// this help the replicator / underlying processor understanding the overall
+		// situation and giveup retrying
+		workerInRetry map[int]struct{}
```
Comment (samarabbas) [inline service/worker/processor.go]: Can we use atomic increment/decrement for keeping the retry count?  Then we don't need  lock

File: service/worker/processor.go [hunk 2]
```diff
@@ -176,44 +260,62 @@ func (p *replicationTaskProcessor) worker(workerWG *sync.WaitGroup) {
...
+			if err != nil && p.isTransientRetryableError(err) {
+				// Keep on retrying transient errors for ever
+				continue ProcessRetryLoop
 			}
-		case <-p.consumer.Closed():
-			p.logger.Info("Consumer closed. Processor shutting down.")
-			return
 		}
+
+		p.updateWorkerStatus(workerID, workerStatusRunning)
```
Comment (samarabbas) [inline service/worker/processor.go]: I think it is much safer to do this as a defer to make sure all code paths which exit set the status back to running.

Comment (wxing1292) [reply service/worker/processor.go]: sure

File: service/worker/processor.go (hunk 1 above)
Comment (wxing1292) [reply service/worker/processor.go]: done

**Pull request merged**